### ISSUE TITLE
Wrong URLs generated in "Tags - Popular" module if any tag's attached to a menu item

### ISSUE THREAD

**Issue by bodlevente** [joomla/joomla-cms#15336]
### Steps to reproduce the issue
Install a basic Joomla! 3.6.5.
Create and publish an article and attach tags to it ("lorem", "ipsum", etc).
Create and publish a "Tags - Popular" module.
Create and publish a menu item with type "Tags » Compact list of tagged items". Select any tag in the menu item's parameters.
On the frontend click on that menu item. (".../index.php/tag-lorem")

### Expected result
In the "Tags - Popular" module the links should be in the basic form like
".../index.php/component/tags/tag/3-ipsum".

### Actual result
In the "Tags - Popular" module the links change to a form if every tag were attached to that menu item.
".../index.php/tag-lorem/3-ipsum".

### System information (as much as possible)
I reproduced the bug with a basic Joomla! 3.6.5 with default settings and with no additional extension.

### Additional comments
I realised this bug when I wanted to use a unique template for a certain tag's "blog view". I realised that after clicking on the menu item every tags' link in the module changes to a form as they were submenus of that menu item, or if the menu item was a kind of blog view of all tags.
(The only solution I found is that to create a hidden menu item listing all tagged items. This way all tags get that menu item's itemid except the one I wanted to attach the unique template.)
I made some investigations and I think that it's not the module that generates the bug. Maybe it's somewhere in the component, in the router that generates the wrong URLs.

**Comment by franz-wohlkoenig** [comment]
I didn't get your Issue: you wrote "Select any tag in the menu item's parameters" and "In the "Tags - Popular" module the links change to a form if every tag were attached to that menu item." 
So i don't understand what's the Bug.

**Comment by bodlevente** [comment]
The bug shows up in the module: the tags' links look like
".../index.php/tag-lorem/3-ipsum"
instead of
".../index.php/component/tags/tag/3-ipsum".
It only happens if we attach a menu item to a tag and we navigate to that menu item (".../index.php/tag-lorem").
(I dont know Joomla so deeply but I think that the links get a wrong ItemID if no menu item attached to them.)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15336">issues.joomla.org/tracker/joomla-cms/15336</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
so your Issue are different URLs using

#### no Menu:
![no menu](https://cloud.githubusercontent.com/assets/8235763/25191278/6b384d40-252f-11e7-86e5-c30fdeddb27d.JPG)

#### Menu:
![menu](https://cloud.githubusercontent.com/assets/8235763/25191294/70e7359e-252f-11e7-8f5c-adff890b8996.JPG)

**Comment by bodlevente** [comment]
No.

I also have taken screenshots.

The first made on the Home page. The upper arrow shows that we are now on the Home page (active menu item --> URL in the browser), the arrow below shows what URL the module contains for the second tag at the moment:

![good](https://cloud.githubusercontent.com/assets/27638302/25198189/05c46418-2547-11e7-8bc7-3926851380c3.png)

The second picture shows the incorrect state of the URLs in the "Tags - Popular" module:

![bad](https://cloud.githubusercontent.com/assets/27638302/25198243/2abc5ad2-2547-11e7-9dd0-a9ab416c05c9.png)

The upper arrow shows that we are now on a menu item with type Tags » Compact list of tagged items (active menu item --> URL in the browser). The only selected tag here at the menu parameters is "lorem". "Ipsum" or any other tag is not included. The other arrow points to the error itself: what kind URL the module contains for the second tag at the moment.

That URL shouldn't change from state "A" to state "B", because it's not the tag "lorem" (that's connected to a menu item) but the tag "ipsum" (that doesn"t have such a connection to any menu item).

**Comment by franz-wohlkoenig** [comment]
@bodlevente Thanks for your Patience, got it.

Set Status in Tracker on "Confirmed".

**Comment by csthomas** [comment]
You can test a solution at #15479

**Comment by zero-24** [comment]
Closing as there is a PR thanks @csthomas !

**Comment by zero-24** [comment]
Please test there if that fixes your issue @bodlevente Thanks!

**Comment by absillitoe** [comment]
I just encountered a similar problem on Joomla 3.7.2. I am not using the popular tags module, however I have several sets of hierarchised tags, one of which is not attached to a menu. I noticed today that the urls for this unattached set of tags are picking up the names of another set of tags that are attached to a menu item. 

I do not know exactly when this problem started, but I have the impression that it must have been introduced during one of the Joomla updates, as I don't remember seeing this problem when I developed the website in question at the end of 2016. However, I cannot be absolutely certain of this.

In order to circumvent this problem, it appears that I have to create menu items for the tags which don't currently have them. These menu items have to be within the same menu as the other tags: it doesn't seem to work if I create a separate hidden menu.

In order to apply the fix referred to here, do I have to apply the changes to the source code referred to here manually? Sorry, I am not very familiar with these "pull requests". 

I would be happy to provide more details of my problem if necessary. 

Thank you.

Alan
